File: Android/sources/androidx/media3/common/ParserException.java
package androidx.media3.common;

import java.io.IOException;

public class ParserException extends IOException {
    public final boolean a;
    public final int b;

    public ParserException(String str, Exception exc, boolean z, int i) {
        super(str, exc);
        this.a = z;
        this.b = i;
    }

    public static ParserException a(String str, RuntimeException runtimeException) {
        return new ParserException(str, runtimeException, true, 1);
    }

    public static ParserException b(Exception exc, String str) {
        return new ParserException(str, exc, true, 4);
    }

    public static ParserException c(String str) {
        return new ParserException(str, (Exception) null, false, 1);
    }

    public final String getMessage() {
        StringBuilder sb = new StringBuilder();
        sb.append(super.getMessage());
        sb.append("{contentIsMalformed=");
        sb.append(this.a);
        sb.append(", dataType=");
        return wj6.l(sb, this.b, "}");
    }
}

File: IOS/v2/iosx/media3/common/ParserException.swift
package iOSx.media3.common;.swift
import java.io.IOException;.swift
public class ParserException extends IOException {.swift
public final boolean a;.swift
public final int b;.swift
public ParserException(String str, Exception exc, boolean z, int i) {.swift
super(str, exc);.swift
this.a = z;.swift
this.b = i;.swift
}.swift
public static ParserException a(String str, RuntimeException runtimeException) {.swift
return new ParserException(str, runtimeException, true, 1);.swift
}.swift
public static ParserException b(Exception exc, String str) {.swift
return new ParserException(str, exc, true, 4);.swift
}.swift
public static ParserException c(String str) {.swift
return new ParserException(str, (Exception) null, false, 1);.swift
}.swift
public final String getMessage() {.swift
StringBuilder sb = new StringBuilder();.swift
sb.append(super.getMessage());.swift
sb.append("{contentIsMalformed=");.swift
sb.append(this.a);.swift
sb.append(", dataType=");.swift
return wj6.l(sb, this.b, "}");.swift
}.swift
}.swift
